subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -166,6 +166,8 @@ public void validateMutation(MutationType type) {
     private boolean dependenciesModified;
     private boolean canBeConsumed = true;
     private boolean canBeResolved = true;
+
+    private boolean canBeMutated = true;
```

Comment:
⭕️ Would it make sense to combine this with the existing resolveState and MutationValidator concepts?